.github/workflows/spell-check.yml

```diff
@@ -0,0 +1,35 @@
+name: Spell Check
+
+on:
+  # Run on-demand using workflow_dispatch
+  workflow_dispatch:
+  # Schedule to run once a week
+  schedule:
+    - cron: '0 0 * * 1' # Runs every Monday at midnight UTC
+
+jobs:
+  spelling:
+    name: Spell check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install pyspellchecker
+        run: |
+          pip install pyspellchecker
+
+      - name: Run spell checker
+        run: |
+          python .github/scripts/spellcheck.py
+
+      - name: Archive the spell check report
+        uses: actions/upload-artifact@v3
```

Comment:
v3 will be deprecated soon (https://github.com/actions/upload-artifact?tab=readme-ov-file#actionsupload-artifact).

Reply:
can you try again did the changes

Reply:
@sm1990 try again